### ISSUE TITLE
Unable to view Nest Login Preferences after generating my own clientId and clientSecret

### ISSUE THREAD

**Issue by frazzlegod** [tonesto7/nest-manager#495]
The "Last Check-In" and "Data Last Received" date/time hadn't been updating for my single Nest  Protect device within SmartThings (integrated using the awesome NST Manager app) for a few days, so I thought it might be down to an issue concerning authenticating with Nest, possibly to do with one of the auth tokens I had been using prior to the need to generate our own auth tokens. @tonesto7 got my auth token working for me originally.

So I took this opportunity to generate my own clientId and clientSecret, as per the recent suggestion relating to auth token issues. I followed the instructions from the wiki page regarding creating my own Nest Developer account, in order to generate my own clientId and clientSecret.

As I already have a Nest account for my Nest Protect device, I logged in with that account and navigated to the Nest Developer console. As per the instructions, I created a new "product", albeit a dummy one just so I could generate the new auth token for myself. I filled out all the fields as instructed and created my new dummy product. Just to reiterate,  just in case the information is useful for diagnosis, my Nest account and Nest Developer account are one and the same thing; I didn't create a separate Developer account as I already had my existing Nest account anyway. 

With my new dummy product created successfully, as per the instructions, I copied the "Product ID" value from my new Nest Developer product to the "clientId" value field in the NST Manager App settings within my SmartThings IDE. Similarly, I copied the "Product Secret" value to the "clientSecret" value field within the SmartThings IDE as well. Clicked Update, Save and then Publish (For me).

Now to the SmartThings app. I navigated to my installation of the NST Manager App and reset the auth token (as I thought this could be the issue why my "Last Check-In" date/time etc. were not updating for my Nest Protect device), which then of course forced me to login to Nest again. I used my same Nest account credentials as before to log in. This did work, and I received the expected "Works with Nest" authorization page, even quoting my own name as requesting the access to Nest, presumably because I set my own name when creating my own Nest Developer dummy product. I clicked the "Authorize" button. That, too, worked. I then received the "SmartThings is now linked to Nest" message. All good so far. 

My problem comes now. When I now go back into the settings screens for the NST Manager SmartApp within the SmartThings app, I notice 2 things:

- Under "Devices & Location", it now states "(null)" for the "Home/Away Status". Previously, it correctly stated "Home".

- Under "Application Preferences", when I tap "Nest Login Preferences" to confirm that my credentials have indeed been taken and I'm in a logged in state, I keep receiving the "Something's Wrong, we can't load your screen right now" page with a "Retry" button, which, when tapped, just reloads that same error page repeatedly. 

Could anyone please help me by throwing some light on why these issues have occurred after I have generated my own auth token and successfully authenticated with it? @tonesto7 @imnotbob

Many thanks in advance, 
frazzlegod

**Comment by frazzlegod** [comment]
Closing issue.

Managed to eventually uninstall NST Manager v5.2.0 from my SmartThings, and re-install latest Nest Manager v5.2.1. I state "eventually" as it was not straightforward at all. After creating my own auth token and authenticating with that, the SmartApp went into a messed up state and refused to uninstall for ages.

But, as I said, I have now managed to uninstall and re-install.

However, I now have a new issue - please see issue #496 that I just opened.

**Comment by tonesto7** [comment]
I can't help make the install process any more 'straightforward' because of SmartThings limitations.

**Comment by tonesto7** [comment]
Also blame Nest for refusing to give us any more tokens

**Comment by frazzlegod** [comment]
Hi @tonesto7,

My issue was primarily with the **UN**install of the previous v5.2.0, rather than the re-install of the latest v5.2.1. This was due to something getting messed up somewhere after I had created my own auth token and updated the clientId and clientSecret in the SmartApps's settings.

I'm not blaming anyone for anything to be honest. I work in the IT industry as a developer and know only too well that things don't always happen as expected; that's the nature of the beast :)

I understand that Nest no longer providing anymore auth tokens to you has been problematic and no doubt a headache for you, but now that you have provided detailed instructions for how to create our own auth tokens going forward, and the fact that I have followed the aforementioned instructions and successfully authenticated with Nest using my own auth token now, has completely helped, and I personally thank you for that.

I love your apps and thank you for all your hard work, time and effort that has gone into creating them. I just had a few teething problems after having created my own auth token, that's all.

Thank you once again for such great and useful apps...keep up the excellent work :)

Regards,
frazzlegod
